src/utils/listen-for-network-messages.ts

```diff
@@ -91,7 +94,16 @@ export function queueEntityStateAsMessage(entityState: EntityState) {
 }
 
 function onEntityStateCreated(response: { data: EntityState[] }) {
-  // console.log("entity_state_saved", response);
+  const rootNid = APP.getSid(response.data[0]!.create_message.networkId);
+  const eid = APP.world.nid2eid.get(rootNid);
+  if (eid) {
+    if (hasComponent(APP.world, Pinnable, eid)) {
+      addComponent(APP.world, Pinned, eid);
+    }
+    if (hasComponent(APP.world, FileInfo, eid)) {
+      FileInfo.flags[eid] |= FILE_INFO_FLAGS.IS_PERMANENT;
+    }
+  }
```

Comment:
If we remove the need to track this derived state separately (`Pinnable`, `Pinned`, and `FileInfo`'s `IS_PERMANENT` flag) as described in a separate comment, then this code can be removed.

If that were not the case, I would encourage us to move the processing of any message data to the `networkReceiveSystem`. Ideally, the only thing happening at this point in the program would be queuing data received from the phoenix channels. We do some parsing and minor revisions to the messages, but try to let the `networkReceiveSystem` be responsible for things that affect the world state.